src/data/repositories/OptionsD2Repository.ts

```diff
@@ -0,0 +1,25 @@
+import { D2Api } from "@eyeseetea/d2-api/2.36";
+import { Id, NamedRef } from "../../domain/entities/Ref";
+import { apiToFuture, FutureData } from "../api-futures";
+import { OptionsRepository } from "../../domain/repositories/OptionsRepository";
+import { Future } from "../../domain/entities/generic/Future";
+
+export class OptionsD2Repository implements OptionsRepository {
+    constructor(private api: D2Api) {}
+
+    get(id: Id): FutureData<NamedRef> {
+        if (!id) return Future.success({ id: "", name: "" });
+        return apiToFuture(
+            this.api.metadata.get({
+                options: { fields: { id: true, name: true }, filter: { id: { eq: id } } },
```

Comment:
Instead of id will be code, right? Because mainSyndromeId, suspectedDiseaseId and notificationSourceId are codes (maybe we can change the name to Code instead of Id?)